ssz/sedes/integer.py

```diff
@@ -6,62 +6,51 @@
 
 class Integer:
     """
-    A sedes for integers (Handles both int<N> and uint<N>).
+    A sedes for integers (uint<N>).
     """
-    # This variable captures 'N' in 'int<N>' or 'uint<N>'
     num_bytes = 0
-    allows_signed_ints = True
 
-    def __init__(self, serializer_type):
-        # Make sure that the serializer_type is string and defined
-        invalid_serializer_msg = "Unvalid serializer type given, string of type 'intN' or 'uintN' expected"  # noqa: E501
-        if serializer_type is None or not isinstance(serializer_type, str):
-            raise SerializationError(invalid_serializer_msg, serializer_type)
-
-        if serializer_type[:3] == 'int':
-            num_bits = int(serializer_type[3:])
-            assert num_bits % 8 == 0
-            self.num_bytes = num_bits // 8
-            self.allows_signed_ints = True
-
-        elif serializer_type[:4] == 'uint':
-            num_bits = int(serializer_type[4:])
-            assert num_bits % 8 == 0
-            self.num_bytes = num_bits // 8
-            self.allows_signed_ints = False
-
-        else:
-            raise SerializationError(invalid_serializer_msg, serializer_type)
+    def __init__(self, num_bytes):
+        self.num_bytes = num_bytes
 
     def serialize(self, val):
         if isinstance(val, bool) or not isinstance(val, int):
-            raise SerializationError('Can only serialize integer values', val)
+            raise SerializationError(
+                'As per specified sedes object, can only serialize non-negative integer values',
+                val
+            )
 
-        if not self.allows_signed_ints:
-            assert val >= 0
+        if val < 0:
+            raise SerializationError(
+                'As per specified sedes object, can only serialize non-negative integer values',
+                val
+            )
+
+        try:
+            serialized_obj = val.to_bytes(self.num_bytes, 'big')
+        except OverflowError as err:
+            raise SerializationError('As per specified sedes object, %s' % err, val)
 
-        return val.to_bytes(self.num_bytes, 'big', signed=self.allows_signed_ints)
+        return serialized_obj
 
-    def _deserialize(self, data, start_idx):
+    def deserialize_from(self, data, start_index):
```

Comment:
Sorry for changing my mind here, but I don't like the `from` now that I see it. It sounds like it refers to `data` instead of the `start_index`. `slice` would be more clear, but it clashes with the Python object. What do you think of `deserialize_segment`?